app/models/hyrax/collection_type.rb

```diff
@@ -3,10 +3,17 @@ class CollectionType < ActiveRecord::Base
     self.table_name = 'hyrax_collection_types'
     validates :title, presence: true, uniqueness: true
     validates :machine_id, presence: true, uniqueness: true
+    before_save :ensure_no_collections
+    before_destroy :ensure_no_collections
```

Comment:
I can't recall, but I think I was doing this as first pass.  So I think the follow on ticket you wrote to be 💯 .